### ISSUE TITLE
GAN - Ch.4 수정안 (comments 모음 티켓)

### ISSUE THREAD

**Issue by SDSTony** [Pseudo-Lab/Tutorial-Book#53]

**Comment by SDSTony** [comment]
- 소제목 입력 부탁드립니다 (4.1 ~~~, 4.2 ~~~ 등등)

- `이전장에서는` -> 이전 장에서는

- `이번장에서는` -> 이번 장에서는

- 4장의 ImageDataset class와 3장의 victorianDataset class가 같은 함수인것 같은데, 함수명을 통일 시키는 것은 어떨까요? @mmminji 
 - imageDataset class에는 gray 이미지를 convert('RGB')로 읽어오고 victorianDataset class에는 convert('L')로 읽어오고 있습니다. 저번 주 회의 때 안건으로 올라온것 같은데 어떤것으로 통일하기로 했었죠..? 참고로 gray파일을 RGB로 읽어오니 channel값이 3으로 들어와서 별도의 reshape없이 imshow가 되는것 같습니다 

- `transform.Normalize에서 Normalize 크기를 지정해줍니다.` -> 해당 문구를 아래 셀로 옮기면 어떨까 합니다 

- Normalize에 입력된 값들이 Victorian400 데이터의 RGB 평균값과 표준편차 값인지 궁금합니다

- `pix2pix의 특징은 일반적인 인코더-디코더 ~~` -> pix2pix의 특징은 Generator에 인코더-디코더 대신 U-NET을 ~~ '

- discriminator에도 skip connection이 들어가 있나요? 

- `패러미터` -> 파라미터 or 인자
-

**Comment by mmminji** [comment]
- transforms_ 에 transforms.Resize는 필요없을 것 같습니다.

- 호열님과 논의하여 3,4장 맞추도록 하겠습니다. (Dataset Class이름, convert("RGB" / "L"), transforms 인자) 

- dataloader 와 val_dataloader 이 현재 똑같습니다. train 하나만 두거나 train/val을 따로 두는 것이 좋을것 같습니다.

**Comment by SDSTony** [comment]
>imageDataset class에는 gray 이미지를 convert('RGB')로 읽어오고 victorianDataset class에는 convert('L')로 읽어오고 있습니다. 저번 주 회의 때 안건으로 올라온것 같은데 어떤것으로 통일하기로 했었죠..? 참고로 gray파일을 RGB로 읽어오니 channel값이 3으로 들어와서 colab에서 별도의 reshape없이 imshow가 되는것 같습니다

모델 input 레이어에 256 256 3 으로 받도록 레이어를 구성해서 convert('RGB')로 합니다 

> victorian400 통계값
gray_mean : tensor([0.5350]) / 
gray_std : tensor([0.1882]) 
color_mean : tensor([0.5809, 0.5269, 0.4568]) / 
color_std : tensor([0.1932, 0.1918, 0.1794])

**Comment by SDSTony** [comment]
2장에서 통곗값 구하는거 설명하고 3장 4장에 normalize할 때 통곗값 사용하기

**Comment by SDSTony** [comment]
> dataloader 와 val_dataloader 이 현재 똑같습니다. train 하나만 두거나 train/val을 따로 두는 것이 좋을것 같습니다.

epoch = 12
val_dataloader 삭제
val_dataloder 대신 dataloader로 검증하기

**Comment by hwaneest** [comment]
![image](https://user-images.githubusercontent.com/33773557/106632453-1a57fe80-65c1-11eb-8db1-6c017f4f9ab5.png)

**Comment by elibooklover** [comment]
> > dataloader 와 val_dataloader 이 현재 똑같습니다. train 하나만 두거나 train/val을 따로 두는 것이 좋을것 같습니다.
> 
> epoch = 12
> val_dataloader 삭제
> val_dataloder 대신 dataloader로 검증하기

epoch = 12 아니라 batch_size = 12를 말씀하시는거죠?

**Comment by SDSTony** [comment]
> epoch = 12 아니라 batch_size = 12를 말씀하시는거죠?

네 batch size = 12 맞습니다 제가 잘못 적었네요 ㅠㅠ
